src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -169,6 +169,8 @@ string GetHeaderSeparator (string name)
 #endif
 		protected override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage request, CancellationToken cancellationToken)
 		{
+			Volatile.Write (ref sentRequest, true);
```

Comment:
I used Volatile following the patterns used in the other handlers to keep everything as close as possible.